src/networking.c

```diff
@@ -1026,6 +1026,18 @@ int clientHasPendingReplies(client *c) {
     }
 }
 
+/* Return true if client connected from loopback interface */
+int localConnection(client *c) {
+    /* unix-socket */
+    if (c->flags & CLIENT_UNIX_SOCKET) return 1;
+
+    /* tcp */
+    char cip[NET_IP_STR_LEN+1] = { 0 };
+    connPeerToString(c->conn, cip, sizeof(cip)-1, NULL);
+
+    return !strcmp(cip,"127.0.0.1") || !strcmp(cip,"::1");
```

Comment:
btw, if we want to can cache this in the client struct so we don't have to repeat that next time, but i don't think it's necessary, considering it protects configs / commands that are not used often.